core/src/main/java/io/github/tcdl/msb/message/MessageFactory.java

```diff
@@ -52,28 +56,33 @@ public Message createBroadcastMessage(Message.Builder messageBuilder, Payload pa
     public Message.Builder createRequestMessageBuilder(String namespace, MessageTemplate messageTemplate, Message originalMessage) {
         Topics topic = new Topics(namespace, namespace + ":response:" +
                 this.serviceDetails.getInstanceId());
-        return createMessageBuilder(topic, messageTemplate, originalMessage);
+        return createMessageBuilder(topic, messageTemplate, originalMessage,false);
     }
 
     public Message.Builder createResponseMessageBuilder(MessageTemplate messageTemplate, Message originalMessage) {
         Topics topic = new Topics(originalMessage.getTopics().getResponse(), null);
-        return createMessageBuilder(topic, messageTemplate, originalMessage);
+        return createMessageBuilder(topic, messageTemplate, originalMessage, true);
     }
 
     public Message.Builder createBroadcastMessageBuilder(String namespace, MessageTemplate messageTemplate) {
         Topics topic = new Topics(namespace, null);
-        return createMessageBuilder(topic, messageTemplate, null);
+        return createMessageBuilder(topic, messageTemplate, null, false);
     }
 
     public Acknowledge.Builder createAckBuilder() {
         return new Acknowledge.Builder().withResponderId(Utils.generateId());
     }
 
-    private Message.Builder createMessageBuilder(Topics topics, MessageTemplate messageTemplate, Message originalMessage) {
```

Comment:
i would remove extra boolean parameter and add call to   messageBuilder.withCorrelationId(createCorrelationId(originalMessage));
or
  messageBuilder.withCorrelationId(createCorrelationId(null)); 
directly to createRequestMessageBuilder/createResponseMessageBuilder

I find this more clear way to understand code

p.s we used such approach few commits before